### PR TITLE
Updating settings to use sequential-storage, bumping minimq and other deps

### LINKED ISSUE #399: Settings are not persistent across reboots
The issue addresses the latest as of writing this main branch 8a4d037cdce1f4500efb0d5f2314526cfda3fee5 commit.
Issue:
After setting the IP address of the device and rebooting it, all the settings stay unchanged.
How to replicate:
Pull the 8a4d037cdce1f4500efb0d5f2314526cfda3fee5 commit and build it. Flash a booster with it.
See the current settings and try updating the IP address of, for example, the broker:
(This device was used before on the v0.5.0 version of the firmware, and it correctly loaded the settings from then):

> list
...
/broker: "10.42.0.1" [default: "10.0.0.2"]
...

> set /broker "10.42.0.5"
Settings saved. Reboot device (platform reboot) to apply.

> list
...
/broker: "10.42.0.5" [default: "10.0.0.2"]
...
>platform reboot

//After rebooting

> list
...
/broker: "10.42.0.1" [default: "10.0.0.2"]
...

The same thing goes for other settings.

I tried doing the same thing with a never before flashed device. The settings were saved correctly the first time, but after the next attempt, they returned to default ones.

Reflashing it again with v0.5.0 fixes the issue, and I'm able to setup things normally.

### PR DESCRIPTION
**Pull request by ryan-summers**
This PR fixes #399 

This PR updates the firmware to utilize `sequential-storage` for keeping settings in flash as opposed to using a single binary blob. This enables us to expand and modify the settings structure going forwards.

This PR also updates various MQTT-related dependencies to the latest versions.

TODO:
- [x] Publish new version of minireq

### CODE REVIEW

File: src/hardware/setup.rs [hunk 1]
```diff
@@ -289,8 +289,15 @@ pub fn setup(
         mac
     };
 
+    let mut flash = {
+        let flash = stm32f4xx_hal::flash::LockedFlash::new(device.FLASH);
+        const SECTOR_SIZE: usize = 128 * 1024;
+        Flash::new(flash, 6 * SECTOR_SIZE)
```
Comment (jordens) [inline src/hardware/setup.rs:295]: I don't get the change from `7*` to `6*` and the doubling of the reserved space size in the linker script. Could you explain?

**Review by jordens — COMMENTED** [review]
Otherwise LGTM

File: src/hardware/setup.rs (hunk 1 above)
Comment (jordens) [reply src/hardware/setup.rs:295]: Ah. Can we write this as `(8 - 2)` instead of `6`? That was my confusion.

Comment (ryan-summers) [reply src/hardware/setup.rs]: FIxed in 7631955a768b033582db8a4a86747d3d2d792d5e

**Comment by jordens** [comment]
Does this also address #380 ?

**Review by jordens — APPROVED** [review]

**Comment by ryan-summers** [comment]
Ah yes, this was on my TODO. I'll take a look

**Review by jordens — APPROVED** [review]

**Pull request merged**